zork++/src/lib/utils/template.rs

```diff
@@ -0,0 +1,199 @@
+            Err(_) => {
+                log::error!("Error create git respotory")
+            }
+        }
+    }
+}
+
+const INTERFACE_MOD_FILE: &'static [u8] = "
+export module math;
+export namespace math {
+    int sum(int num1, int num2);
+    int multiply(int num1, int num2);
+    int substract(int num1, int num2);
+    int divide(int num1, int num2);
+}
+"
+.as_bytes();
+
+const SRC_MOD_FILE: &'static str = "
+module math;
+// Implementation of the definitions on the module unit interface
+// for the sum and multiply math operations
+namespace math {
+    int sum(int num1, int num2) {
+        return num1 + num2;
+    }
+    int multiply(int num1, int num2) {
+        return num1 * num2;
+    }
+}
+";
+
+const SRC_MOD_FILE_2: &'static str = "
+module math;
+// Implementation of the definitions on the module unit interface
+// for the substract and divide math operations
+namespace math {
+    int substract(int num1, int num2) {
+        return num1 - num2;
+    }
+    int divide(int num1, int num2) {
+        return num1 / num2;
+    }
+}
+";
+
+const MAIN: &'static str = r#"
+import std;
+import math;
+int main() {
+    std::cout << "Hello from an autogenerated Zork project!" << std::endl;
+    std::cout << "RESULT '+': " << math::sum(2, 8) << std::endl;
+    std::cout << "RESULT '-': " << math::substract(8, 2) << std::endl;
+    std::cout << "RESULT '*': " << math::multiply(2, 8) << std::endl;
+    std::cout << "RESULT '/': " << math::divide(2, 2) << std::endl;
+    return 0;
+}
+"#;
+
+const ZORK_CONF: &'static str = r#"
+#This file it's autogenerated as an example of a Zork config file
+[project]
+name = "calculator"
+authors = [ "Zero Day Code" ]  # Replace this for the real authors
+[compiler]
+cpp_compiler = "clang"
+cpp_standard = "20"
+std_lib = "libcpp"
+[build]
+output_dir = "./out"
+[executable]
+executable_name = "<autogenerated_executable>"
+sources = [
+    "*.cpp"
+]
+auto_execute = true
+[tests]
+tests_executable_name = "zork_proj_tests"
+sources = [
+    "*.cpp"
+]
+auto_run_tests = true
+[modules]
+base_ifcs_dir = "<project_name>/ifc/"
+interfaces = [ "*.cppm" ]
+base_impls_dir = "<project_name>/src/"
+implementations = [
+    "math.cpp",
+    "math2.cpp=[math]"
+]
+"#;
+
+fn create_file<'a>(path: &PathBuf, name_file: &'a str, buff_write: &'a [u8]) {
+    let mut file = File::create(path.join(name_file)).expect(
+        format!(
+            "Error create in directory: {}, file: {}",
+            path.to_str().unwrap(),
+            name_file
+        )
+        .as_str(),
+    );
+    file.write_all(buff_write).expect(
+        format!(
+            "Error write in directory: {}, file: {}",
+            path.to_str().unwrap(),
+            name_file
+        )
+        .as_str(),
+    );
+}
+
+fn create_directory(path_create: &Path) {
+    DirBuilder::new()
+        .recursive(true)
+        .create(path_create)
+        .expect(format!("Error create directory: {}", path_create.to_string_lossy()).as_str())
```

Comment:
`.to_string_lossy()`. Try to use one of the `UTF-8` alternatives to mantain consistency across the whole API